### PR TITLE
Update dependencies and fix javadoc artifacts

### LINKED ISSUE #94: Fix javadoc artifacts
Currently, the javadoc artifacts published to Maven central don't have any content. We should follow the steps referenced here to publish real javadocs: https://kotlinlang.org/docs/dokka-gradle.html#build-javadoc-jar

### PR DESCRIPTION
**Pull request by pkwarren**
Update to the latest versions of spotless, android libraries, and the Maven plugin. Fix javadoc generation for Maven artifacts (previously empty).

Fixes #94.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -51,11 +51,23 @@ allprojects {
     spotless {
         isEnforceCheck = false // Disables lint on gradle builds.
         kotlin {
-            ktlint().editorConfigOverride(mapOf("ktlint_experimental" to "enabled"))
+            ktlint().editorConfigOverride(
+                mapOf(
+                    "ktlint_experimental" to "enabled",
+                    // Disabled until https://github.com/pinterest/ktlint/pull/2273 is released
+                    "ktlint_standard_function-expression-body" to "disabled",
+                    ),
```
Comment (rebello95) [inline build.gradle.kts:59]: nit: indentation

File: build.gradle.kts [hunk 2]
```diff
@@ -51,11 +51,23 @@ allprojects {
     spotless {
         isEnforceCheck = false // Disables lint on gradle builds.
         kotlin {
-            ktlint().editorConfigOverride(mapOf("ktlint_experimental" to "enabled"))
+            ktlint().editorConfigOverride(
+                mapOf(
+                    "ktlint_experimental" to "enabled",
+                    // Disabled until https://github.com/pinterest/ktlint/pull/2273 is released
+                    "ktlint_standard_function-expression-body" to "disabled",
```
Comment (pkwarren) [inline build.gradle.kts:58]: This is a new experimental rule. In some cases it does the right thing but without the fix above it leads to compile errors. It also is rewriting a large number of files - I might just leave this disabled and manually update the simpler methods to follow the recommended style.

File: build.gradle.kts [hunk 3]
```diff
@@ -76,16 +88,8 @@ allprojects {
     }
     tasks.withType<DokkaTask>().configureEach {
         dokkaSourceSets.configureEach {
-            reportUndocumented.set(false)
             skipDeprecated.set(true)
             jdkVersion.set(8)
-            perPackageOption {
-                matchingRegex.set("build\\.buf.*")
-                suppress.set(true)
```
Comment (pkwarren) [inline build.gradle.kts:84]: This was causing docs not to be included.

File: protoc-gen-connect-kotlin/src/main/kotlin/com/connectrpc/protocgen/connect/internal/Parameters.kt [hunk 4]
```diff
@@ -42,7 +42,8 @@ internal fun parse(input: String): Configuration {
     val parameters = parseGeneratorParameter(input)
     return Configuration(
         generateCallbackMethods = parameters[CALLBACK_SIGNATURE]?.toBoolean() ?: false,
-        generateCoroutineMethods = parameters[COROUTINE_SIGNATURE]?.toBoolean() ?: true, // Defaulted to true.
+        // Defaulted to true.
+        generateCoroutineMethods = parameters[COROUTINE_SIGNATURE]?.toBoolean() ?: true,
```
Comment (rebello95) [inline protoc-gen-connect-kotlin/src/main/kotlin/com/connectrpc/protocgen/connect/internal/Parameters.kt:46]: IMO it was clearer before, but if this violates a lint rule it's not a big deal

File: build.gradle.kts [hunk 5]
```diff
@@ -76,16 +88,8 @@ allprojects {
...
-            }
-        }
-        if (name == "dokkaGfm") {
-            outputDirectory.set(project.file("${project.rootDir}/docs/3.x"))
```
Comment (pkwarren) [inline build.gradle.kts:88]: Not sure where this is coming from - we don't publish GFM anywhere so I've switched to the standard `dokkaHtml` format.

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -6,17 +6,17 @@ dokka = "1.9.0"
...
 okio = "3.6.0"
 protobuf = "3.24.4"
 slf4j = "2.0.9"
-spotless = "6.21.0"
+spotless = "6.22.0"
 
 [libraries]
 android = { module = "com.google.android:android", version.ref = "android" }
-android-material = { module = "com.google.android.material:material", version = "1.9.0" }
+android-material = { module = "com.google.android.material:material", version = "1.10.0" }
```
Comment (pkwarren) [inline gradle/libs.versions.toml:19]: This requires a new Android SDK.

**Review by rebello95 — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (pkwarren) [reply build.gradle.kts]: Fixed - forgot to re-run the linter.

File: protoc-gen-connect-kotlin/src/main/kotlin/com/connectrpc/protocgen/connect/internal/Parameters.kt (hunk 4 above)
Comment (pkwarren) [reply protoc-gen-connect-kotlin/src/main/kotlin/com/connectrpc/protocgen/connect/internal/Parameters.kt:46]: Yeah this failed a new lint rule that required comments to be on their own lines.

**Pull request merged**